server/devdocs/test/index.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import fs from 'fs';
+import fspath from 'path';
+import request from 'superagent';
+import mockery from 'mockery';
+import { expect } from 'chai';
+
+/**
+ * Test setup
+ */
+mockery.registerMock( 'config', {
+	isEnabled: function() {
+		return true;
+	}
+} );
+mockery.registerMock( 'server/devdocs/search-index', {
+	index: {} // for speed - the real search index is very large
+} );
+
+/**
+ * Internal dependencies
+ */
+import devdocs from '../';
+
+/**
+ * Module variables
+ */
+function getDocument( base, path, cb ) {
+	if ( typeof path === 'function' ) {
+		cb = path;
+		path = base;
+	} else {
+		// This mimics what the browser does in this situation
+		path = fspath.join( base, path );
+	}
+	request.get( 'http://localhost:9993/devdocs/service/content' )
+		.query( { path: path } )
+		.end( ( error, res ) => {
+			cb( error, res );
+		} );
+}
+
+describe( 'devdocs server', () => {
+	let app, server;
+
+	before( done => {
+		app = devdocs();
+		server = app.listen( 9993, done );
```

Comment:
Would require restructuring `server/devdocs/index.js`.  Possible, but I'm not convinced it is worth doing here:

- There are several different error conditions that have different HTTP response codes.
- It would be inconsistent (and I think it would make the code more confusing) to restructure one route, but not the others.